lib/services/API/country_institute_api.dart

```diff
@@ -0,0 +1,64 @@
+import 'package:mobile_app/constants.dart';
+import 'package:mobile_app/models/failure_model.dart';
+import 'package:mobile_app/utils/api_utils.dart';
+
+abstract class CountryInstituteAPI {
+  Future<dynamic> getCountries(String query);
+  Future<dynamic> getInstitutes(String query);
+}
+
+class HttpCountryInstituteAPI implements CountryInstituteAPI {
+  List<dynamic> data;
+  static const String COUNTRY = 'country';
```

Comment:
I think now we don't require these constants, you can remove them.

Reply:
I think we will need them. I have used them in `cv_typeahead_field.dart` file. 
```
    if (toggle == HttpCountryInstituteAPI.COUNTRY) {
        return await countryInstituteObject.getCountries(
         pattern,
        );
    }
    if (toggle == HttpCountryInstituteAPI.EDUCATIONAL_INSTITUTE) {
        return await countryInstituteObject.getInstitutes(
            pattern,
        );
    }
```

This implementation allows us to use the same typeahead field class in other places as well.

Reply:
Ok then please move into the `cv_typeahewd_field.dart` since it's no longer used in API.

Reply:
I have made the changes accordingly.